src/account/UpgradeableModularAccount.sol

```diff
@@ -298,14 +296,35 @@ contract UpgradeableModularAccount is
         AccountStorage storage _storage = getAccountStorage();
 
         ModuleEntity sigValidation = ModuleEntity.wrap(bytes24(signature));
+        signature = signature[24:];
 
-        (address module, uint32 entityId) = sigValidation.unpack();
         if (!_storage.validationData[sigValidation].isSignatureValidation) {
-            revert SignatureValidationInvalid(module, entityId);
+            (address _module, uint32 _entityId) = sigValidation.unpack();
+            revert SignatureValidationInvalid(_module, _entityId);
+        }
+
+        ModuleEntity[] memory preSignatureValidationHooks =
+            getAccountStorage().validationData[sigValidation].preValidationHooks;
+
+        for (uint256 i = 0; i < preSignatureValidationHooks.length; ++i) {
+            (address hookModule, uint32 hookEntityId) = preSignatureValidationHooks[i].unpack();
+
+            bytes memory currentSignatureSegment;
+
+            (currentSignatureSegment, signature) = signature.advanceSegmentIfAtIndex(uint8(i));
+
+            // If this reverts, bubble up revert reason.
+            IValidationHookModule(hookModule).preSignatureValidationHook(
+                hookEntityId, msg.sender, hash, currentSignatureSegment
+            );
         }
 
+        signature = signature.getFinalSegment();
+
+        (address module, uint32 entityId) = sigValidation.unpack();
```

Comment:
Nit: move this line up above 301 and remove 302?

Reply:
Ended up being unnecessary, as the changes from SMA pulled this out into a separate function.